### PR TITLE
Added support for `ls -l --color` to color symlink targets as well.

### PR DESCRIPTION
**Pull request by mtsoltan**
The feature works, and I'm pretty happy with it, however,

I'm not entirely sure about the way the test is structured. It tests the colors of all of:
* Symlinks to existing files
* Symlinks to existing dirs
* Symlinks to non-existing files
* Symlinks to non-existing dirs
* Symlinks to relative dirs, and to absolute dirs.
using `ls -laR --color` so it's a bit intricate.

It's also written in a tool-agnostic / environment-agnostic method (unlike `test_ls_color` which is tailored for our version of `ls`) such that the GNU `ls` will also pass this test (even though it will fail `test_ls_color`), and such that `ls` running with a different LS_COLORS variable set will still pass this test (even though it will fail `test_ls_color`).

I tried to make it as clear as possible by abstracting away the low-level string manipulation and option handling in local functions.

Also, I don't like the fact that it uses the built `mkdir`, `touch`, and `ln` to create the test-case. **My justification is**: the stuff in the `scene.fixture` just do not provide the same functionality as `ln` because they always resolve all paths to absolute paths, so my dependency on `ln` cannot be discarded yet. I used the coreutils `ln` because I have no guarantees that a system-level `ln` will exist on windows. That means I only have to use `ln`, why am I using `mkdir` and `touch` too? To be consistent, I didn't want to have illegible mixed code that uses both the fixture and a coreutils tool.

### CODE REVIEW

**Comment by mtsoltan** [comment]
So far, clippy just gave an error on an unnecessary borrow (which was a warning on my local machine, but is an error here for some reason), and everything else passed.

Let me know what you think, so I can fix all of it when I'm on again, then push.

**Comment by miDeb** [comment]
This change looks good to me. Thanks!
> which was a warning on my local machine, but is an error here for some reason

That's because we treat warnings as errors here.

> Also, I don't like the fact that it uses the built mkdir, touch, and ln to create the test-case. 

That's alright.

**Comment by mtsoltan** [comment]
The merge conflicts introduced yesterday could safely be resolved in favor of `ls-symlink-colors`.

They are introduced by e1d7c85d76f667d7c5ddebb7ee9a5e2e538258a5.

File: tests/by-util/test_ls.rs [hunk 1]
```diff
@@ -335,27 +335,13 @@ fn test_ls_long() {
...
-        }
-    }
-
     let scene = TestScenario::new(util_name!());
     let at = &scene.fixtures;
     at.mkdir(&at.plus_as_string("test-long-dir"));
     at.touch(&at.plus_as_string("test-long-dir/test-long-file"));
     at.mkdir(&at.plus_as_string("test-long-dir/test-long-dir"));
 
     for arg in &["-l", "--long", "--format=long", "--format=verbose"] {
```
Comment (miDeb) [inline tests/by-util/test_ls.rs:344]: Could you fix this unused variable warning?

**Comment by mtsoltan** [comment]
Passes all checks, no warnings.

The warning was a logic issue in which this would initialize functions on windows when the entire test (tests differences in groups and unames) should be not(windows) only.

**Comment by miDeb** [comment]
Thanks!

**Pull request merged**